### ISSUE TITLE
Буквица в firefox

### ISSUE THREAD

**Issue by alkorlos** [web-standards-ru/web-standards.ru#107]
В firefox буквица отображается неправильно
![first-letter](https://user-images.githubusercontent.com/14668230/85205523-e0393580-b32c-11ea-999f-b53f8f7da8c3.jpg)
Исправить не получилось, похоже что это баг реализации ::first-letter в firefox. При одинаковых стилях разное расположение псевдоэлемента.

Если обернуть первый символ в span с классом first-letter и дать ему стили от ::first-letter буквица отображается корректно во всех браузерах.

**Comment by pepelsbey** [comment]
Да, наверное стоит написать фильтр для 11ty, который будет первую букву первого параграфа заворачивать в нужно.

**Comment by dom1n1k** [comment]
> похоже что это баг реализации ::first-letter в firefox

Игнорируется line-height.
https://bugzilla.mozilla.org/show_bug.cgi?id=290125
(особого внимания заслуживает дата баг-репорта)

Либо оборачивать первую букву в `span`, либо накостылить нечто такое:
```
@supports (-moz-appearance: none) {
    .content > p:first-of-type::first-letter {
        margin-top: 10px;  /* some magic number */
    }
}
```

**Comment by pepelsbey** [comment]
Ой-ей, я бы лучше избежал любых браузерных хаков :)

**Comment by alkorlos** [comment]
Проверил ещё раз, действительно в firefox line-height игнорируется, но без line-height поведение в firefox и chrome у ::first-letter разное. Видимо нет кроссбраузерного поведения не только у line-height, но и у других свойств, либо наследование свойств работает по разному. Так что надежнее, до исправления багов, отказаться от ::first-letter.

**Comment by PetePearl** [comment]
я починю

**Comment by pepelsbey** [comment]
Что-то меня всё больше тянет на фильтр, который первую букву оборачивает в спан с классом. Раз уж у нас такая кроссбраузерность и такой дизайн.

**Comment by firefoxic** [comment]
Хотел было поразбираться с этой проблемой. Открыл первую (свежайшую) статью для инспектирования. И увидев такое:
![image](https://user-images.githubusercontent.com/3382798/104798497-642aa180-57d8-11eb-8d03-9c64ef6d695a.png)
не удержался и попробовал удалить всего одно слово из первого абзаца:
![image](https://user-images.githubusercontent.com/3382798/104798473-36ddf380-57d8-11eb-9c05-33944fff6e5b.png)
И сразу возникает вопрос: А нужна ли эта буквица? Кажется с ней одни только проблемы: кроссбраузерность страдает, а в случае с абзацем в одну строку ещё ломает второй абзац. Я обычно не придираюсь к самой концепции буквицы (ну нравится может людям, пусть используют), но в этом примере слово «Спор» вообще выглядит как отдельные слова «С пор».

Но если таки буквица нужна, то тогда второй вопрос:
Какой она должна быть высоты? Верхний край буквицы должен соответствовать верхнему краю прописной или строчной букве в первой строке? В основном в первой строке редко встречаются прописные, и нынешняя буквица выглядит непонятно к чему привязанной по высоте. В редких случаях, вроде [этой статьи](https://web-standards.ru/articles/inherit-initial-unset-revert/), сразу после буквицы идут прописные буквы, и тогда более-менее органично смотрится. Но это скорее исключение, чем правило.
В макете верх буквицы уходит выше линии 8-пиксельной сетки, а прописные буквы первой строки — ниже этой линии.

**Comment by firefoxic** [comment]
Есть ещё один недочёт. Статья вполне может начинаться с чего-то обёрнутого в `<code>`. Из его содержимого первый символ будет не просто визуально оторван от остальной части, но и насильно переформатирован в Dewi Expanded. Но это ещё не всё: буквица у нас не апперкейсится, из-за чего если первый символ в нижнем регистре (что в кусках кода чаще всего), то он обесценивает наш хак для обхода бага в Firefox с недействующим `line-height` в `::first-letter`.
Вот пример в Chromium:
![image](https://user-images.githubusercontent.com/3382798/104893811-f2dc2180-5984-11eb-81aa-0ce1ab7e1813.png)
Он же в Firefox:
![image](https://user-images.githubusercontent.com/3382798/104893950-1ef7a280-5985-11eb-951f-cd075c6828e5.png)

`uppercase` конечно не долго прописать, но как это скажется на восприятии текста, ведь мы привыкли и хотим видеть куски кода такими, какими пишем их в редакторе кода…

**Comment by pepelsbey** [comment]
> И сразу возникает вопрос: А нужна ли эта буквица?

Я понимаю, что самый простой способ решить задачу — это её отменить. И это даже правильный подход, сам так часто делаю. Но всё-таки буквица нам нужна, это сильный элемент, обозначающий начало статьи.

> Статья вполне может начинаться с чего-то обёрнутого в `<code>`

У нас не просто так редакторы есть, они такого не допустят, всегда можно переформулировать.

**Comment by alkorlos** [comment]
Выше приводили ссылку на [[CSS21] floated :first-letter pseudo-element should act like normal inline (e.g., support line-height)](https://bugzilla.mozilla.org/show_bug.cgi?id=290125), репорт о баге буквицы в firefox. Его открыли восемнадцать лет назад, закрыли шесть месяцев назад, в firefox 110 должны быть эти изменения.
Но в актуальном firefox 114 этот баг все еще есть, в актуальном firefox nightly 116.0a1 этой проблемы нет.
В firefox nightly 113.0a1 этой проблемы тоже нет. Похоже в nightly уже какое то время есть исправление этого бага, но в актуальный релиз они не включаются.

Знал что баг, на который ссылка выше, починили, хотел закрыть это issue, но не получилось. Не понятно разработчики firefox не включают исправление этого бага в стабильной версии браузера или это уже другая проблема.

**Comment by pepelsbey** [comment]
<img width="286" alt="Image" src="https://github.com/user-attachments/assets/bebcd426-2b77-49af-8c51-091d93db1e06" />

Всё как будто хорошо в Firefox, пять лет спустя)

**Comment by alexeyten** [comment]
Само починилось 😂
